app/src/main/java/com/onewelcome/showcaseapp/feature/userregistration/twostepregistration/TwoStepVerificationScreen.kt

```diff
@@ -0,0 +1,163 @@
+package com.onewelcome.showcaseapp.feature.userregistration.twostepregistration
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavController
+import com.onewelcome.core.components.ShowcaseCard
+import com.onewelcome.core.components.ShowcaseTopBar
+import com.onewelcome.core.theme.Dimensions
+import com.onewelcome.showcaseapp.R
+import androidx.compose.runtime.LaunchedEffect
+import com.onewelcome.showcaseapp.feature.userregistration.twostepregistration.TwoStepRegistrationViewModel.NavigationEvent
+import com.onewelcome.showcaseapp.feature.userregistration.twostepregistration.TwoStepRegistrationViewModel.UiEvent
+import com.onewelcome.showcaseapp.navigation.Screens
+import kotlinx.coroutines.flow.Flow
+
+@Composable
+fun TwoStepVerificationScreen(
+  navController: NavController, pinNavController: NavController
+) {
+  val parentEntry = remember(navController) {
+    navController.getBackStackEntry(Screens.TwoStepRegistration.route)
+  }
+  val viewModel: TwoStepRegistrationViewModel = hiltViewModel(parentEntry)
+  ListenForNavigationEvents(viewModel.navigationEvents, pinNavController, navController)
+  TwoStepVerificationScreenContent(challengeCode = viewModel.uiState.challengeCode, onNavigateBack = {
+    viewModel.onEvent(UiEvent.CancelRegistration)
```

Comment:
onNavigateBack = {
      viewModel.onEvent(UiEvent.CancelRegistration)
      navController.popBackStack()
    }

we're canceling registration and navigating back just one screen back, to the `optionalDataSubmitScreen`, instead of the `twoStepRegistrationScreen`